### ISSUE TITLE
New request: PsychonautWiki

### ISSUE THREAD

**Issue by sihma** [openzim/zim-requests#634]
- Website URL: https://psychonautwiki.org/wiki/Main_Page
- License: CC-BY-SA 4.0
- Desired ZIM Title: PsychonautWiki
- Desired ZIM Description: The open encyclopedia of psychonautics
- Desired ZIM Icon –png (URL or attach one): ![eye](https://github.com/openzim/zim-requests/assets/82542273/3fa07799-6664-4393-bc94-608fecf9cf8d)
- Language (ISO 639-3): eng
- Is this a MediaWiki?: yes

**Comment by arrall** [comment]
Bump.
Looking for this as well.

**Comment by arrall** [comment]
@kelson42 Sorry about the duplicate. 
I wasn't sure if this just got buried and forgotten about.
Is there a plan to add this one at some point?
What is the general ETA for getting a request filled?

**Comment by Popolechien** [comment]
@arrall wikis are complicated at the moment but we have the tooling to at least try. If the scraping fails, there usually are two reasons:
- the mediawiki instance is not up to date and it's up to the owners to fix it (if they can/are still around);
- our own mediawiki offliner is off its rocker and the issue will be addressed in an ucoming fix (bearing in mind that the project has not started yet as we are still seeking funding).

Bottom line: wikis we can try pretty quickly, but if it don't work then there's no clear ETA. But we'll keep the issue open as long as it is not resolved.

**Comment by kelson42** [comment]
@Popolechien Please create recipe and let me know if problems

**Comment by Popolechien** [comment]
@kelson42 I see they have a sign requesting people use [their API ](https://api.psychonautwiki.org/) rather than directly crawl the site. Is this something MWoffliner's `API path` will handle?

**Comment by Popolechien** [comment]
@kelson42 Created https://farm.openzim.org/recipes/psychonautwiki_en_all but it failed.

**Comment by RavanJAltaie** [comment]
The recipe is failing with this error:
```
												"POST",
												"PROPFIND",
												"PROPPATCH",
												"PURGE",
												"PUT",
												"REBIND",
												"REPORT",
												"SEARCH",
												"SOURCE",
												"SUBSCRIBE",
												"TRACE",
												"UNBIND",
												"UNLINK",
												"UNLOCK",
												"UNSUBSCRIBE"
											],
											"STATUS_CODES": {
												"100": "Continue",
												"101": "Switching Protocols",
												"102": "Processing",
												"103": "Early Hints",
												"200": "OK",
												"201": "Created",
												"202": "Accepted",
												"203": "Non-Authoritative Information",
												"204": "No Content",
												"205": "Reset Content",
												"206": "Partial Content",
												"207": "Multi-Status",
												"208": "Already Reported",
												"226": "IM Used",
												"300": "Multiple Choices",
												"301": "Moved Permanently",
												"302": "Found",
												"303": "See Other",
												"304": "Not Modified",
												"305": "Use Proxy",
												"307": "Temporary Redirect",
												"308": "Permanent Redirect",
												"400": "Bad Request",
												"401": "Unauthorized",
												"402": "Payment Required",
												"403": "Forbidden",
												"404": "Not Found",
												"405": "Method Not Allowed",
												"406": "Not Acceptable",
												"407": "Proxy Authentication Required",
												"408": "Request Timeout",
												"409": "Conflict",
												"410": "Gone",
												"411": "Length Required",
												"412": "Precondition Failed",
												"413": "Payload Too Large",
												"414": "URI Too Long",
												"415": "Unsupported Media Type",
												"416": "Range Not Satisfiable",
												"417": "Expectation Failed",
												"418": "I'm a Teapot",
												"421": "Misdirected Request",
												"422": "Unprocessable Entity",
												"423": "Locked",
												"424": "Failed Dependency",
												"425": "Too Early",
												"426": "Upgrade Required",
												"428": "Precondition Required",
												"429": "Too Many Requests",
												"431": "Request Header Fields Too Large",
												"451": "Unavailable For Legal Reasons",
												"500": "Internal Server Error",
												"501": "Not Implemented",
												"502": "Bad Gateway",
												"503": "Service Unavailable",
												"504": "Gateway Timeout",
												"505": "HTTP Version Not Supported",
												"506": "Variant Also Negotiates",
												"507": "Insufficient Storage",
												"508": "Loop Detected",
												"509": "Bandwidth Limit Exceeded",
												"510": "Not Extended",
												"511": "Network Authentication Required"
											},
											"maxHeaderSize": 16384,
											"globalAgent": {
												"_events": {},
												"_eventsCount": 2,
												"defaultPort": 80,
												"protocol": "http:",
												"options": {
													"noDelay": true,
													"path": null
												},
												"requests": {},
												"sockets": {},
												"freeSockets": {},
												"keepAliveMsecs": 1000,
												"keepAlive": false,
												"maxSockets": null,
												"maxFreeSockets": 256,
												"scheduling": "lifo",
												"
```

**Comment by Popolechien** [comment]
> Please create recipe and let me know if problems

@kelson42 ping

**Comment by benoit74** [comment]
Update to test with `dev` image (more or less 1.14). Also remove all useless settings, this Mediawiki instance seems to be configured with "all defaults" settings.

**Comment by benoit74** [comment]
ZIM built has broken images on all pages : https://dev.library.kiwix.org/viewer#psychonautwiki_en_all_maxi/

Looks like `dev` still has some issues ^^

**Comment by benoit74** [comment]
ZIMs are ready for review at https://dev.library.kiwix.org/#lang=&q=psychonaut

@Popolechien can you please have a look and move them to prod if OK?

**Comment by benoit74** [comment]
@Popolechien up

**Comment by Popolechien** [comment]
Layout seems not to be captured at all. Besides the landing page, compare https://psychonautwiki.org/wiki/Psychoactive_substance_index and https://dev.library.kiwix.org/viewer#psychonautwiki_en_all_nopic_2025-03/A/Psychoactive_substance_index

With a secondary issue whereby code is shown:

![Image](https://github.com/user-attachments/assets/cd7b8592-ef94-4434-8815-3045cd7a3205)

**Comment by benoit74** [comment]
Indeed, still not ready

**Comment by kelson42** [comment]
So, the quality of the scrape improves, see https://dev.library.kiwix.org/viewer#psychonautwiki_en_all_maxi_2025-06/Psychoactive_substance_index ... But a few metadata should at least be fixed ... and maybe then it is ready to publish?

**Comment by benoit74** [comment]
> But a few metadata should at least be fixed

Which metadata? I don't get it

**Comment by kelson42** [comment]
![image](https://github.com/user-attachments/assets/b9bafd34-96d8-4828-bd10-a29c1d8c5e77)

**Comment by benoit74** [comment]
This LGTM

**Comment by CodeF53** [comment]
Whats holding this issue at the moment?

**Comment by Popolechien** [comment]
Nothing but the fact it was old. I've created a [CMS entry](psychonautwiki.org_en_all) and restarted the zimfarm.

**Comment by Popolechien** [comment]
Run failed: https://farm.openzim.org/pipeline/19975e3a-d644-48e7-b1a9-89b294dde149/debug

**Comment by benoit74** [comment]
The website explicitly refuses to grant access to their API ; we need their consent and a technical solution to be allowed to create a ZIM, nothing we can solve scraper side.

See for instance content of https://psychonautwiki.org/w/api.php?action=query&format=json&prop=info%7Credirects%7Crevisions&rdlimit=max&rdnamespace=0&rdprop=pageid%7Ctitle%7Cfragment&formatversion=2&maxlag=5&rawcontinue=true&generator=allpages&gapfilterredir=nonredirects&gaplimit=max&gapnamespace=0 

This is the query we run to get list of articles in the wiki.
